src/main/java/io/cdap/plugin/gcp/bigquery/action/AbstractBigQueryActionConfig.java

```diff
@@ -16,12 +16,36 @@
 
 package io.cdap.plugin.gcp.bigquery.action;
 
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
 import io.cdap.cdap.etl.api.FailureCollector;
 import io.cdap.plugin.gcp.common.GCPConfig;
 
+import javax.annotation.Nullable;
+
 /**
  * Base class for Big Query action configs.
  */
 public abstract class AbstractBigQueryActionConfig extends GCPConfig {
+  public static final String DATASET_PROJECT_ID = "datasetProject";
+
+  @Name(DATASET_PROJECT_ID)
+  @Macro
+  @Nullable
+  @Description("The project in which the dataset specified in the `Dataset Name` is located or should be created."
```

Comment:
@itsankit-google The UI property is Dataset, nit Dataset Name. Should we change the UI or change the description here?

Reply:
So, basically BigQueryExecute Plugin UI had it as `Dataset Name`. So to keep it consistent in all BQ Action plugins, I changed it to `Dataset Name`.

Reply:
Do we also need to say something like "Do we need to add this sentence to the description?  “BigQuery Data Viewer role on this project must be granted to the specified service account to read BigQuery data from this project.”